### PR TITLE
Desk Toys and Lava Lamps

### PR DESCRIPTION
**Pull request by alsoandanswer**
some useless desk fillers

dreamseeker_GesrgibNbd

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1011,3 +999,48 @@
 	throwforce = 1
 	drop_sound = 'sound/items/drop/cardboardbox.ogg'
 	pickup_sound = 'sound/items/pickup/cardboardbox.ogg'
+
+
+//baystation desk toys
+
+/obj/item/toy/desk
+	var/on = 0
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1007]:
```suggestion
	var/on = FALSE
```

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1011,3 +999,48 @@
...
+	var/activation_sound = /decl/sound_category/switch_sound
+
+/obj/item/toy/desk/update_icon()
+	if(on)
+		icon_state = "[initial(icon_state)]-on"
+	else
+		icon_state = "[initial(icon_state)]"
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1014]:
```suggestion
		icon_state = initial(icon_state)
```

File: code/game/objects/items/devices/lighting/lamp.dm [hunk 3]
```diff
@@ -23,10 +22,51 @@
...
+	icon_state = "lavalamp"
+	brightness_on = 3
+	matter = list(DEFAULT_WALL_MATERIAL = 250, MATERIAL_GLASS = 200)
+
+/obj/item/device/flashlight/lamp/lava/update_icon()
+	if(on)
+		set_light(brightness_on)
+	else
+		set_light(0)
+	overlays.Cut()
```
Comment (Geevies) [inline code/game/objects/items/devices/lighting/lamp.dm:38]:
```suggestion
	cut_overlays()
```

File: code/game/objects/items/devices/lighting/lamp.dm [hunk 4]
```diff
@@ -23,10 +22,51 @@
...
+	var/image/I = image(icon = icon, icon_state = "lavalamp-[on ? "on" : "off"]")
+	I.color = light_color
+	overlays += I
```
Comment (Geevies) [inline code/game/objects/items/devices/lighting/lamp.dm:41]:
```suggestion
	add_overlay(I)
```

File: code/game/objects/items/devices/lighting/lamp.dm [hunk 5]
```diff
@@ -23,10 +22,51 @@
...
+	desc = "A kitchy green decorative light."
+	light_color = COLOR_GREEN
+
+/obj/item/device/flashlight/lamp/lava/orange
+	desc = "A kitchy orange decorative light."
+	light_color = COLOR_ORANGE
+
+/obj/item/device/flashlight/lamp/lava/purple
+	desc = "A kitchy purple decorative light."
+	light_color = COLOR_PURPLE
```
Comment (Geevies) [inline code/game/objects/items/devices/lighting/lamp.dm:65]:
```suggestion
	light_color = COLOR_PURPLE

```

**Review by Geevies — CHANGES_REQUESTED** [review]

**Review by fernerr — COMMENTED** [review]
I personally feel most of these seem rather out of place, the dippy bird especially feeling like a meme. Something like lava lamps I'd rather see reflavored/tweaked to fit in better, like a 'elyran plasma lamp' or something.

**Review by Geevies — APPROVED** [review]

**Comment by skull132** [comment]
Agreed with Ferner on this one. Cool idea, but would love some more specific-to-our-setting flavour in it. Also, not enough map integration (read: any) to be relevant.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**